api/routes/books.js

```diff
@@ -46,33 +59,82 @@ router.post('/', async (req, res, next) => {
 router.patch('/:id/reserve', async (req, res, next) => {
   const { id } = req.params
   try {
+    const token = req.headers.authorization.split('Bearer ')[1]
+    if(!token){
```

Comment:
You'll want to first validate that the token is correct before determining whether or not to allow them to perform this action.